src/test/test-seccomp.c

```diff
@@ -322,6 +322,42 @@ static void test_protect_sysctl(void) {
         assert_se(wait_for_terminate_and_check("sysctlseccomp", pid, WAIT_LOG) == EXIT_SUCCESS);
 }
 
+static void test_protect_syslog(void) {
+        pid_t pid;
+
+        log_info("/* %s */", __func__);
+
+        if (!is_seccomp_available()) {
+                log_notice("Seccomp not available, skipping %s", __func__);
+                return;
+        }
+        if (geteuid() != 0) {
+                log_notice("Not root, skipping %s", __func__);
+                return;
+        }
+
+        pid = fork();
+        assert_se(pid >= 0);
+
+        if (pid == 0) {
+#ifdef __NR_syslog
+                assert_se(syscall(__NR_syslog, -1, NULL, 0) < 0);
+                assert_se(errno == EINVAL);
+#endif
+
+                assert_se(seccomp_protect_syslog() >= 0);
+
+#ifdef __NR_syslog
+                assert_se(syscall(__NR_syslog, 0, 0, 0) < 0);
+                assert_se(errno == EPERM);
```

Comment:
maybe also check that open("/dev/kmsg", O_RDONLY) fails with EPERM and didn't before. (some minor additional care should be taken though: these CI tests might run in container environments, where the device is generally not available, hence if the opening didn't work beforehand, then just skip this open() check)

Reply:
I've made all the other changes, but not sure about this one. Should this check really be in the seccomp tests? `seccomp_protect_syslog()` is not going to  restrict access there.

Reply:
sure, test for that can be elsewhere, but it should be there, after all the path /dev/kmsg was misspelled...